21.service/main.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func handler(w http.ResponseWriter, r *http.Request) {
+	s := r.URL.String()
+	d, err := os.Open(s[1:])
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer d.Close()
+	entries, err := d.Readdir(-1)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
this function service each user call separate, you no need to crash entire program in case of one call - just log an error and return the error message to the client